docs/aws/audit/cloudtrailmonitoring/rules/cloudtrail_global_services_logging_duplicated_remediation.mdx

```diff
@@ -1,6 +1,297 @@
+
+3. **Enable Log File Integrity Validation:**
+   - In the CloudTrail console, select your trail.
+   - Enable the "Enable log file validation" option. This feature helps ensure the integrity of your log files and can help identify any duplicate entries.
+
+4. **Review and Consolidate Trails:**
+   - Regularly review your existing trails in the CloudTrail console.
+   - Consolidate multiple trails that might be capturing the same events into a single trail where possible. This reduces the chances of duplicate log entries.
+
+By following these steps, you can effectively prevent duplicate entries in your CloudTrail logs using the AWS Management Console.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent duplicate entries in CloudTrail logs using AWS CLI, you can follow these steps:
+
+1. **Create a New CloudTrail Trail:**
+   Ensure you have a single, well-configured CloudTrail trail to avoid duplicate entries. Use the following command to create a new trail:
+
+   ```sh
+   aws cloudtrail create-trail --name my-trail --s3-bucket-name my-trail-bucket
+   ```
+
+2. **Enable Log File Validation:**
+   Enable log file validation to ensure the integrity of the log files and prevent duplicates. Use the following command:
+
+   ```sh
+   aws cloudtrail update-trail --name my-trail --enable-log-file-validation
+   ```
+
+3. **Ensure Only One Trail is Logging for Each Region:**
+   Verify that you have only one trail logging events for each region. List all trails and check their configurations:
+
+   ```sh
+   aws cloudtrail describe-trails
+   ```
+
+   If you find multiple trails logging for the same region, consider consolidating them.
+
+4. **Enable Multi-Region Trail:**
+   If you need to log events from multiple regions, enable the multi-region trail option to avoid creating separate trails for each region:
+
+   ```sh
+   aws cloudtrail update-trail --name my-trail --is-multi-region-trail
+   ```
+
+By following these steps, you can prevent duplicate entries in CloudTrail logs using AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent duplicate entries in AWS CloudTrail logs using Python scripts, you can follow these steps:
+
+### 1. Set Up AWS SDK for Python (Boto3)
+First, ensure you have the AWS SDK for Python (Boto3) installed. You can install it using pip if you haven't already:
+
+```bash
+pip install boto3
+```
+
+### 2. Create a Python Script to Check for Duplicate Entries
+You can create a Python script that will check for duplicate entries in CloudTrail logs and take appropriate actions to prevent them.
+
+### 3. Use Boto3 to Interact with CloudTrail
+Here is a sample Python script that demonstrates how to interact with CloudTrail and check for duplicate entries:
+
+```python
+import boto3
+import hashlib
+
+# Initialize a session using Amazon CloudTrail
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>